### PR TITLE
Fixed nn.functional.pad constant mode

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Fixed nn.functional.pad constant mode
- Updated tests

The issue was with the fact that `UNARY_POINTWISE` does not properly handle bdim > 0. The fix is to use a macro which explicitly calls `moveBatchDimToFront`.

Related to #240

### CODE REVIEW

**Comment by vfdev-5** [comment]
@zou3519 please review this PR as well. Thanks

**Comment by Chillee** [comment]
Sorry, I don't understand why UNARY_POINTWISE doesn't support bdim > 0 🤔 

Oh, is it specifically that for `constant_pad_nd` it screws up the padding argument?

**Review by Chillee — APPROVED** [review]

**Review by zou3519 — APPROVED** [review]

**Comment by zou3519** [comment]
> Sorry, I don't understand why UNARY_POINTWISE doesn't support bdim > 0 🤔
> 
> Oh, is it specifically that for `constant_pad_nd` it screws up the padding argument?

yes. The padding argument is a list of pads for the last dimensions. For example, `constant_pad_nd(x: Tensor[2, 3], pad=[2, 3, 4, 5])` would pad dimension 0 using (2, 3) and dimension 1 using (4, 5).

Now, if your batch dimension is in the middle (e.g. `constant_pad_nd(x: Tensor[2, B, 3], pad=[2, 3, 4, 5])`) then we need to first move it to the front so it doesn't get padded by mistake
